test_util/test_runner.py

```diff
@@ -135,9 +136,15 @@ def integration_test(
         '-e', 'DNS_SEARCH='+dns_search,
         '-e', 'AWS_ACCESS_KEY_ID='+aws_access_key_id,
         '-e', 'AWS_SECRET_ACCESS_KEY='+aws_secret_access_key,
-        '-e', 'AWS_REGION='+region,
-        '--net=host', '--name='+test_container_name, 'py.test', 'py.test',
-        '-vv', ci_flags, '/integration_test.py']
+        '-e', 'AWS_REGION='+region]
+    for key, value in add_env.items():
+        extra_env = ['-e', key+'='+value]
+        test_env.extend(extra_env)
+    docker_opts = [
+        'docker', 'run', '-v', test_dir+'/integration_test.py:/integration_test.py',
+        '--net=host', '--name='+test_container_name]
+    test_args = ['py.test', 'py.test', '-vv', ci_flags, '/integration_test.py']
+    test_cmd = docker_opts + test_env + test_args
```

Comment:
Rather than splitting this into three parts, why not just do a:
docker_cmd = ['...']
# add_env
# append docker opts

I don't see a good reason for pulling this into three different variables than re-composing them into the overal command.